File: ocmsdk/src/main/java/com/gigigo/orchextra/core/domain/entities/article/base/ArticleButtonSize.java
package com.gigigo.orchextra.core.domain.entities.article.base;

import java.io.Serializable;

public enum ArticleButtonSize implements Serializable {
  BIG("big"),
  MEDIUM("medium"),
  SMALL("small");

  private final String size;

  ArticleButtonSize(String size) {
    this.size = size;
  }

  public String getSize() {
    return size;
  }

  public static ArticleButtonSize convertStringToEnum(String size) {
    ArticleButtonSize[] values = ArticleButtonSize.values();
    for (ArticleButtonSize value : values) {
      if (value.getSize().equals(size)) {
        return value;
      }
    }

    return MEDIUM;
  }
}

File: ocmsdk/src/main/java/com/gigigo/orchextra/core/domain/entities/article/base/ArticleButtonSize.kt
package com.gigigo.orchextra.core.domain.entities.article.base

import java.io.Serializable

enum class ArticleButtonSize constructor(val size: String) : Serializable {
  BIG("big"),
  MEDIUM("medium"),
  SMALL("small");

  companion object {
    fun convertFromString(size: String?): ArticleButtonSize {
      val values = ArticleButtonSize.values()
      for (value in values) {
        if (value.size == size) {
          return value
        }
      }
      return MEDIUM
    }
  }
}